deploy-asm.sh

```diff
@@ -33,40 +33,30 @@ echo "No CLUSTERNAME variable set"
 exit
 fi
 
-gcloud services enable cloudtrace.googleapis.com
-
-gcloud container clusters get-credentials $CLUSTERNAME --project=$PROJECT --zone=$LOCATION
-kubectl config set-context $CLUSTERNAME
-
-echo "Installing ASM..."
-curl -s https://storage.googleapis.com/csm-artifacts/asm/asmcli_1.13 > asmcli
-chmod +x asmcli
+echo "Register GKE cluster to the fleet..."
+gcloud container fleet memberships register $CLUSTERNAME \
+    --gke-cluster $LOCATION/$CLUSTERNAME \
+    --enable-workload-identity
+    
+echo "Enable Managed ASM on the project..."
+gcloud container fleet mesh enable
 
 sleep 5s
 
-./asmcli install \
-  --project_id $PROJECT \
-  --cluster_name $CLUSTERNAME \
-  --cluster_location $LOCATION \
-  --fleet_id $PROJECT \
-  --output_dir asmoutput \
-  --enable_all \
-  --ca mesh_ca
+echo "Enable Managed ASM on the GKE cluster..."
+gcloud container fleet mesh update \
+    --management automatic \
+    --memberships $CLUSTERNAME
 
-sleep 10s
+sleep 10s # we need to wait more here...
```

Comment:
Can you just increase the sleep then? How long does it normally take? 
Is there a way for us to check before we proceed?

Reply:
Yeah, sorry, I didnt' explicitly mentioned that, but I'm trying to find a way to automate that part, hopefully with a smart `kubectl` or `gcloud` command and wait on it... not sure but that will be very much better if possible.